### DISCUSSION TITLE
Can I conditionally call useWatch?

### DISCUSSION THREAD

**Discussion by devinrhode2** [react-hook-form/react-hook-form#3624] [Q&A]
I have a requirement where submit button needs to be valid once all the fields have valid inputs. If you are going through the form, and have entered everything correctly, when you are entering in the last field of the form, I need to be able to watch() and, _the moment that all form values are valid_, enable the form's Submit button.

I have a function which only calls watch() as a last resort, when you are on the last input field, however, it would be nice to pass down a function which the final consumer calls to determine if the submit button is enabled or not. Of course, passing watch would be a fruitless endeavor here, the idea is of course to pass a useWatch based callback, which determines submit enabled state.

Which leads me to my question... can I "conditionally" call useWatch? I could return `boolean | function` but I'd rather just return my same utility, `isSubmitEnabled`, and have the consumer always call it, however, the utility may only sometimes call useWatch.

I believe it wouldn't work, since hooks can't be called conditionally.

~However, I could wire up some things to permit conditionally _returning_ a hook instead of the literal boolean.~ Actually, I think no matter how I go about it, in some sense, at some point, I will conditionally calling a hook.

**Comment by devinrhode2** [comment]
Alternatively, it would be _really nice_ if RHF had a function, like `.trigger()`, but one which _didn't actually cause errors to be shown_. 

Currently I just get a Record/map of all input fields `validator` callbacks (my own construct) and, get the result from `watch()` (no args) and then call each fields `validator` to see if they are all valid. Inside my code I've named this essentially something like "areAllFieldsValid"

**Comment by bluebill1049** [comment]
hey @devinrhode2 is that just `isValid` formState?

**Comment by devinrhode2** [comment]
No, isValid doesn't seem to work, since I need to activly watch all form data for errors, and isValid is tied to mode, and I can't really change mode

**Reply by bluebill1049** [reply]
what's not working? what's the mode you are using? do you have a codesandbox to share?

**Comment by devinrhode2** [comment]
Using mode onTouched, so real rhf validation doesn't happen until user has
blurred a field.

**Reply by bluebill1049** [reply]
can you attach onChagne with trigger?

**Comment by spoeken** [comment]
Could you seperate it into another component and conditionally display that component?
Like this component where I'm utilising useWatch: https://codesandbox.io/s/react-hook-form-auto-save-xgulp?file=/src/AutoSave.tsx

**Comment by elissaioskon** [comment]
I had the same issue. What I did is that I used `watch` conditionally. 


```javascript
 const minDate = condition && watch('dataFrom');
```
